addons/web/static/src/js/views/abstract_view.js

```diff
@@ -183,84 +179,140 @@ var AbstractView = Factory.extend({
-                controlPanelModel,
                 fields: this.fields,
                 searchMenuTypes: params.searchMenuTypes,
                 view: this.fieldsView,
-                views: action.views && action.views.filter(v => v.multiRecord === this.multi_record),
+                views: params.action.views && params.action.views.filter(
+                    v => v.multiRecord === this.multi_record
+                ),
                 withBreadcrumbs: params.withBreadcrumbs,
                 withSearchBar: params.withSearchBar,
             };
-            this.controllerParams.controlPanelModel = controlPanelModel;
-            this.controllerParams.controlPanelProps = controlPanelProps;
+            this.controllerParams.controlPanel = {
+                Component: ControlPanelComponent,
+                props: controlPanelProps,
+            };
         }
 
+        // Search panel params
         if (this.withSearchPanel) {
-            this.searchPanelParams = {
-                arch: (params.controlPanelFieldsView || {}).arch,
+            // Search panel (Model)
+            const SearchPanelComponent = this.config.SearchPanel;
+            const defaultValues = {};
+            Object.keys(this.loadParams.context).forEach((key) => {
+                let match = /^searchpanel_default_(.*)$/.exec(key);
+                if (match) {
+                    defaultValues[match[1]] = this.loadParams.context[key];
+                }
+            });
+            extensions[SearchPanelComponent.modelExtension] = {
+                archNodes: searchPanelInfo.children,
                 defaultNoFilter: params.searchPanelDefaultNoFilter,
-                fields: this.fields,
-                model: this.loadParams.modelName,
-                state: controllerState.spState,
+                defaultValues,
+            };
+            this.controllerParams.withSearchPanel = true;
+            this.rendererParams.withSearchPanel = true;
+            // Search panel (Component)
+            const searchPanelProps = {
+                scrollTop: params.controllerState ?
+                    params.controllerState.scrollState.searchPanel :
+                    0,
+            };
+            if (searchPanelInfo.attrs.class) {
+                searchPanelProps.className = searchPanelInfo.attrs.class;
+            }
+            this.controllerParams.searchPanel = {
+                Component: SearchPanelComponent,
+                props: searchPanelProps,
             };
         }
-    },
 
-    //--------------------------------------------------------------------------
-    // Public
-    //--------------------------------------------------------------------------
+        const searchModel = new ActionModel(extensions, {
+            env: Component.env,
+            modelName: params.modelName,
+            actionContext: Object.assign({}, this.loadParams.context), // not sure we should use loadParams.context!
```

Comment:
maybe actionContext should not be called that way (see init)?
or maybe params.context might be used only in tests?

Reply:
the comment is still valid

Reply:
Changed to `actionDomain` > `domain` and `actionContext` > `context`